@xen-orchestra/backups/_cleanVm.js

```diff
@@ -134,9 +134,55 @@ const listVhds = async (handler, vmDir) => {
       )
   )
 
-  return { vhds, interruptedVhds }
+  return { vhds, interruptedVhds, aliases }
 }
 
+async function checkAliases(aliasPaths, targetDataRepository, { handler, onLog = noop, remove = false }) {
+  const aliasFound = []
+  for (const path of aliasPaths) {
+    const target = await resolveAlias(handler, path)
+
+    if (!isVhdFile(target)) {
+      onLog(`Alias ${path} references a non vhd target:  ${target}`)
+      if (remove) {
+        await handler.unlink(target)
+        await handler.unlink(path)
+      }
+      continue
+    }
+
+    try {
+      const { dispose } = await openVhd(handler, target)
+      dispose()
+    } catch (e) {
+      onLog(`target ${target} of alias ${path} is missing or broken`)
+      if (remove) {
+        try {
+          await VhdAbstract.unlink(handler, path)
+        } catch(e){}
```

Comment:
Can we avoid trying to delete if the VHD is missing?

This would avoid an unnecessary operation and would avoid hiding a possibly interesting error.